clustered/client/src/test/java/org/ehcache/clustered/client/internal/config/xml/ClusteringServiceConfigurationParserTest.java

```diff
@@ -424,6 +443,227 @@ public void testServersWithClusterTierManagerAndOptionalPorts() throws Exception
+    URI connectionUri = new URI("terracotta://localhost:9510/my-application");
+    ClusteringServiceConfiguration serviceConfig = ClusteringServiceConfigurationBuilder.cluster(connectionUri)
+      .timeouts(Timeouts.DEFAULT)
+      .expecting()
+      .defaultServerResource("main")
+      .build();
+    Configuration clusterConfiguration = getClusterConfiguration(serviceConfig);
+    ConfigurationTranslator configurationTranslator = new ConfigurationTranslator(clusterConfiguration);
+    String retValue = configurationTranslator.toXml();
+    System.out.println(retValue);
+    String startTag = "<ehcache:config xmlns:ehcache=\"http://www.ehcache.org/v3\"><ehcache:service>";
+    String endTag = "</ehcache:service></ehcache:config>";
+    String entireString = startTag + retValue + endTag;
+    System.out.println(entireString);
+    XmlConfiguration config = new XmlConfiguration(makeConfig(new String[] { entireString }));
+
+    assertThat(config.getServiceCreationConfigurations(), is(notNullValue()));
+    assertThat(config.getServiceCreationConfigurations(), iterableWithSize(equalTo(1)));
+
+    ClusteringServiceConfiguration retServiceConfig =
+      (ClusteringServiceConfiguration)config.getServiceCreationConfigurations().iterator().next();
+
+    assertThat(connectionUri, equalTo(retServiceConfig.getClusterUri()));
+    assertThat(true, not(retServiceConfig.isAutoCreate()));
+    assertThat(retServiceConfig.getServerConfiguration(), is(notNullValue()));
+
+    assertThat(retServiceConfig.getTimeouts(), is(notNullValue()));
+    assertThat(retServiceConfig.getTimeouts().getConnectionTimeout(), equalTo(Timeouts.DEFAULT.getConnectionTimeout()));
+    assertThat(retServiceConfig.getTimeouts()
+      .getReadOperationTimeout(), equalTo(Timeouts.DEFAULT.getReadOperationTimeout()));
+    assertThat(retServiceConfig.getTimeouts().getConnectionTimeout(), equalTo(Timeouts.DEFAULT.getConnectionTimeout()));
+
+    Map<String, ServerSideConfiguration.Pool> retPoolMap = retServiceConfig.getServerConfiguration().getResourcePools();
+    assertThat(retPoolMap, is(notNullValue()));
+    assertThat(0, is(equalTo(retPoolMap.size())));
+  }
+
+  @Test
+  public void testTranslateServiceCreationConfigurationWithNoServerSideConfig() throws Exception {
+    URI connectionUri = new URI("terracotta://localhost:9510/my-application");
+    ClusteringServiceConfiguration serviceConfig = ClusteringServiceConfigurationBuilder.cluster(connectionUri)
+      .timeouts(Timeouts.DEFAULT)
+      .build();
+    Configuration clusterConfiguration = getClusterConfiguration(serviceConfig);
+    ConfigurationTranslator configurationTranslator = new ConfigurationTranslator(clusterConfiguration);
+    String retValue = configurationTranslator.toXml();
+    System.out.println(retValue);
+    String startTag = "<ehcache:config xmlns:ehcache=\"http://www.ehcache.org/v3\"><ehcache:service>";
+    String endTag = "</ehcache:service></ehcache:config>";
```

Comment:
Can you please refactor the start. end tag mechanic into a private method?

Reply:
Sure, wll do it